### PR TITLE
feat: Add support for Xamarin TV.OS

### PR DESCRIPTION
**Pull request by glennawatson**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
This adds support to TV.OS

**What is the current behavior? (You can also link to an open issue here)**
No TV.OS support.

**What is the new behavior (if this is a feature change)?**
Have changed the projects, event builder.

**What might this PR break?**
Not tested 100% with the TVOS, but would only be the user group.

**Other information**:
Thanks @emmanuele-villa -- if you can test this at some point that would be awesome 👍

### CODE REVIEW

**Comment by emmanuele-villa** [comment]
@glennawatson are you sure about the src/EventBuilder/Platforms/tvOS.cs correct dll path?
On my mac  I have a generic "Xamarin.iOS.framework" that contains the folders mono/Xamarin.iOS, mono/Xamarin.TVOS and mono/Xamarin.WatchOS, so the path on my machine is:

@"/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.TVOS10/Xamarin.TVOS10.dll";

I will test it for sure whenever is ready or I can manage to compile it.. I will ask for support for that in another thread :)

File: src/EventBuilder/Platforms/tvOS.cs [hunk 1]
```diff
@@ -14,7 +14,7 @@ public TVOS(string referenceAssembliesLocation)
         {
             if (PlatformHelper.IsRunningOnMono()) {
                 var assembly =
-                    @"/Library/Frameworks/Xamarin.TVOS10.framework/Versions/Current/lib/mono/Xamarin.TVOS10/Xamarin.TVOS10.dll";
+                    @"/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.TVOS10/Xamarin.TVOS10.dll";
```
Comment (glennawatson) [inline src/EventBuilder/Platforms/tvOS.cs]: Thanks for that.

File: src/EventBuilder/Program.cs [hunk 2]
```diff
@@ -109,6 +109,10 @@ private static void Main(string[] args)
                         platform = new UWP();
                         break;
 
+                    case AutoPlatform.TVOS:
+                        platform = new UWP();
```
Comment (ghuntley) [inline src/EventBuilder/Program.cs]: This needs to be changed to `new TVOS` (or whatever the class is called)

**Comment by glennawatson** [comment]
How's your compilation etc going @emmanuele-villa ?

**Comment by ghuntley** [comment]
I haven't run this code yet; did a quick 30 second eyeball for anything obvious. The `tvos` team has been created and `CODEOWNERS` in this PR was updated by me.

**Comment by emmanuele-villa** [comment]
@ghuntley sure, I just changed the path because I'm trying to compile it on my machine right now :)
@glennawatson I added my mac server ip in the build.cake and finally I'm able to compile after wiping and reinstalling visual studio!

**Comment by emmanuele-villa** [comment]
@glennawatson so, I managed to add the built dll to a sample apple tv project: now I can find the tvos specific class (e.g. ReactiveViewController) but I have a missing method exception at runtime.
Now... I've added a xamarintvos10 folder and dll to ReactiveUI and Splat in the packages folder, but I honestly don't know if that's the correct way to test it, because:
- I have problem importing base ReactiveUI (circular depencency error)
- I dont know where to put the System.Reactive dll because I cant find the apple specific folder of it in the packages

Long story short: I can now use the tvos base classes; it crashes at runtime but I'm not sure if that's my fault at wrongly importing the dlls or what

**Comment by glennawatson** [comment]
A good way I've found in the past to simulate a build from our products is to setup a nuget reference to the path, 

So the build.cmd script will generate .nupkg in the artifacts folder. 

Then in Visual Studio 2017 you can go to 

1. Go to **Tools** menu -> **Options**
2. Search for nuget, Select **NuGet Package Manager** -> **Package Sources**
3. Click the green + button on the top right of the options screen.
4. Then for the name enter "rxui testing"
5. Then for the Source, you can press the "..." next to the source textbox to navigate to the artifacts founder.

Then you can use the nuget package manager like you would other nuget package, just in the dialog select "rxUI Testing" as your source.

That way you get more of a "realistic" testing environment.

**Comment by emmanuele-villa** [comment]
thanks @glennawatson !
I still have some strange "dll not found" warnings, but I'm now able to use .WhenAnyValue without exceptions in a tvos freshly new project.

**Comment by glennawatson** [comment]
Can you copy/paste those DLL warnings in to the PR? Curious what you are getting.

**Comment by emmanuele-villa** [comment]
Those are the errors:

> Could not find `System.Threading.Tasks.Extensions` referenced by assembly `System.Reactive, Version=4.1.0.0, Culture=neutral, PublicKeyToken=94bc3704cddfc263`.
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.XamForms' (culture: '')
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.XamForms' (culture: '')
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.Winforms' (culture: '')
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.Winforms' (culture: '')
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.Wpf' (culture: '')
> Xamarin.TVOS: Unable to locate assembly 'ReactiveUI.Wpf' (culture: '')

For the first one, I have added manually the System.Threading.Tasks.Extensions package with no luck. Also, I noticed that when I added System.Reactive.Linq to the packages, the local System.Reactive dll swapped automatically with the online nuget one, if that helps

**Comment by emmanuele-villa** [comment]
@glennawatson back at work I noticed that I also have those warnings (except for the first one) also on the ios online nuget, so I guess that's ok

**Comment by glennawatson** [comment]
Hmm fair enough. Want to lodge a issue on GitHub about the warnings? Will see if we can get someone with more iOS experience to take a look. I will submit the PR once I get another review from someone.

**Comment by glennawatson** [comment]
Hi @reactiveui/ios-team this project is almost identical to your project with just a few controls removed. Check out and give some feedback. Thanks.
